### PR TITLE
Update all the things

### PR DESCRIPTION
**Pull request by XhmikosR**
* Drop Node.js < 14 support
* Remove debug dependency
* Switch to GitHub Actions CI
* Add CodeQL workflow
* Add c8 for coverage
* Update all dependencies
* Fix tests on Windows

TODO:

- [x] update module-lookup-amd to v8.0.0

### CODE REVIEW

**Comment by XhmikosR** [comment]
@mrjoelkemp this is ready from my side. Please have a look and let me know!

**Comment by XhmikosR** [comment]
@mrjoelkemp friendly ping :)

**Comment by XhmikosR** [comment]
@mrjoelkemp friendly ping again :)

I believe the changes here will help the project in the long term, I'm just not 100% on the TS change I needed to make.

Otherwise I will go ahead and merge this later.

**Review by mrjoelkemp — APPROVED** [review]

**Comment by XhmikosR** [comment]
@mrjoelkemp https://github.com/dependents/node-filing-cabinet/pull/119#discussion_r1174928345

I had to wrap the call to `resolveJSModule` in try/catch because newer versions throw. The fallback (`|| result`) might be moot now, but I left them there for the time being.

File: index.js [hunk 1]
```diff
@@ -253,15 +253,17 @@ function tsLookup({dependency, filename, directory, webpackConfig, tsConfig, tsC
     result = namedModule.resolvedModule.resolvedFileName;
     if (namedModule.resolvedModule.extension === '.d.ts' && noTypeDefinitions) {
       const resolvedFileNameWithoutExtension = result.replace(namedModule.resolvedModule.extension, '');
-      result = ts.resolveJSModule(resolvedFileNameWithoutExtension, path.dirname(filename), host) || result;
+      try {
```
[the remark this answers is not in the document]
Comment (mrjoelkemp) [reply index.js:249]: Should we add a debug call in the catch for a debugging hint that the TS compiler threw an exception?

**Comment by mrjoelkemp** [comment]
Should we add a debug call in the catch for a debugging hint that the TS compiler threw an exception?

**Comment by XhmikosR** [comment]
Sure thing. How do you want the debug call to look like? Include any further info or just a message?

**Comment by XhmikosR** [comment]
```
CABINET 980: Given filename: C:\Users\xmr\Desktop\detective\node-filing-cabinet\test\fixtures\ts\index.ts
CABINET 980: which has the extension: .ts
CABINET 980: found a resolver for .ts
CABINET 980: performing a typescript lookup
CABINET 980: given typescript config: undefined
CABINET 980: no tsconfig given, defaulting
CABINET 980: processed typescript config: undefined
CABINET 980: processed typescript config type: undefined
CABINET 980: with options: {}
CABINET 980: ts.resolveJSModule threw an Error: Could not resolve JS module 'C:/Users/xmr/Desktop/detective/node-filing-cabinet/test/fixtures/ts/withOnlyTypeDef' starting at 'C:\Users\xmr\Desktop\detective\node-filing-cabinet\test\fixtures\ts'. Looked in: C:/Users/xmr/Desktop/detective/node-filing-cabinet/test/fixtures/ts/withOnlyTypeDef.js, C:/Users/xmr/Desktop/detective/node-filing-cabinet/test/fixtures/ts/withOnlyTypeDef.jsx, C:/Users/xmr/Desktop/detective/node-filing-cabinet/test/fixtures/ts/withOnlyTypeDef/package.json, C:/Users/xmr/Desktop/detective/node-filing-cabinet/test/fixtures/ts/withOnlyTypeDef/index.js, C:/Users/xmr/Desktop/detective/node-filing-cabinet/test/fixtures/ts/withOnlyTypeDef/index.jsx
CABINET 980: result: C:/Users/xmr/Desktop/detective/node-filing-cabinet/test/fixtures/ts/withOnlyTypeDef.d.ts
CABINET 980: resolved path for ./withOnlyTypeDef.d.ts: C:\Users\xmr\Desktop\detective\node-filing-cabinet\test\fixtures\ts\withOnlyTypeDef.d.ts
          √ still returns the .d.ts file if no JS file is found
```

**Pull request merged**